### PR TITLE
DAOS-13292 control: Use cart API to detect fabric

### PR DESCRIPTION
**Pull request by kjacque**
- Add a lib/hardware package to collect fabric interface information through CART API.
- Remove custom OFI and UCX packages and dependencies.
- Update Go githook to ignore deleted files.

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/control/lib/hardware/cart/cart.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	classLibFabric = "ofi"
+	classUCX       = "ucx"
+	classNA        = "na"
+)
+
+// crtFabricDevice is a single fabric device discovered by CART.
+type crtFabricDevice struct {
+	Class    string
+	Protocol string
+	Device   string
```
Comment (mjmac) [inline src/control/lib/hardware/cart/cart.go:29]: Is there any possibility that this struct may eventually be dumped as JSON? If so, probably a good idea to just go ahead and add the field tags now.

**Review by mjmac — COMMENTED** [review]
Looks good to me... Much simpler to have a single provider.

File: src/control/lib/hardware/cart/cart.go (hunk 1 above)
Comment (kjacque) [reply src/control/lib/hardware/cart/cart.go:29]: I could add them. It seems unlikely we'd use it outside this package though.

Comment (kjacque) [reply src/control/lib/hardware/cart/cart.go]: Fixed

**Comment by kjacque** [comment]
Test failure is https://daosio.atlassian.net/browse/DAOS-15598

**Review by knard-intel — DISMISSED** [review]
LGTM

**Review by tanabarr — APPROVED** [review]
nice work

**Comment by kjacque** [comment]
Test failure is [DAOS-15686](https://daosio.atlassian.net/browse/DAOS-15686). I'm going to exclude the affected test and re-run.

**Review by knard-intel — DISMISSED** [review]

**Review by tanabarr — DISMISSED** [review]

**Review by mjmac — APPROVED** [review]

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

**Review by knard-intel — APPROVED** [review]
LGTM
The functional test failing should not be related to this PR and to be a transient error corrupting the json output.

**Review by tanabarr — APPROVED** [review]

**Pull request merged**